subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -772,25 +780,28 @@ const struct bt_mesh_model_cb _bt_mesh_scene_srv_cb = {
 static int scene_setup_srv_init(struct bt_mesh_model *model)
 {
 	struct bt_mesh_scene_srv *srv = model->user_data;
+	struct bt_mesh_model *dtt_srv;
+	int err;
 
 	if (!srv) {
 		return -EINVAL;
 	}
 
 	srv->setup_mod = model;
 
-	/* Model extensions:
-	 * To simplify the model extension tree, we're flipping the
-	 * relationship between the scene server and the scene
-	 * setup server. In the specification, the scene setup
-	 * server extends the scene server, which is the opposite of
-	 * what we're doing here. This makes no difference for the mesh
-	 * stack, but it makes it a lot easier to extend this model, as
-	 * we won't have to support multiple extenders.
-	 */
-	bt_mesh_model_extend(srv->model, srv->setup_mod);
+	dtt_srv = bt_mesh_model_find(bt_mesh_model_elem(model),
```

Comment:
Additional check is required here. If DTT server is not found on this element, then all element numerically smaller than this element should be checked and DTT instance on the largest element should be used. 

Ref: Section 3.3.3.1 in MshMDLv1.0.1

Reply:
According to 5.3.4.1, MeshMDLv1.0.1, Scene server may only be present on the primary element of the node. So for _the current spec_ this check is enough.